xbmc/pvr/windows/GUIViewStatePVR.cpp

```diff
@@ -77,8 +77,10 @@ void CGUIViewStateWindowPVRGuide::SaveViewState(void)
 
 CGUIViewStateWindowPVRTimers::CGUIViewStateWindowPVRTimers(const int windowId, const CFileItemList& items) : CGUIViewStatePVR(windowId, items)
 {
-  AddSortMethod(SortByLabel, 551, LABEL_MASKS("%L", "%I", "%L", ""));   // FileName, Size | Foldername, empty
-  AddSortMethod(SortByDate, 552, LABEL_MASKS("%L", "%J", "%L", "%J"));  // FileName, Date | Foldername, Date
+  SortAttribute sortAttributes(CSettings::Get().GetBool("filelists.ignorethewhensorting") ? SortAttributeIgnoreArticle : SortAttributeNone);
+  sortAttributes = (SortAttribute)((int)sortAttributes | SortAttributeIgnoreFolders);
```

Comment:
@ksooo The (int) cast isn't strictly necessary to compile, only the (SortAttribute) one. I used c casts for both simply to keep the same formatting as the other examples where a similar construction was used. 

Unlike your comment on the sortAttributes declaration which I already included, this change although better cpp, makes the code less consistent. I'm happy to take my share of new contributer beastings, but I thought you weren't interested in list sorting or bikesheds... ;)

If there is a desire to 'fix' this, imo it would be better to overload the |= operator to cope with the SortAttribute enum (or similar) so no cast is needed at all. There is however the question of the SortAttribute enum being used as a bitmask, where other code uses #defines for this (e.g. PVR core in the PVR_TIMER.iAttributes code).

I'm not yet confident enough in 'proper cpp' or 'the way Kodi is supposed to do things' to take on this re-factoring task myself. I also tend to agree with @opdenkamp that re-factoring the sorting code is out of scope for a simple 3 line PVR only PR. 

Imho it's better to keep consistency of usage here and re-factor separately.
Does this approach work for you?